tests/unit/core.js

```diff
@@ -818,6 +818,21 @@ exports.testES6TemplateLiteralMultiline = function (test) {
   test.done();
 };
 
+exports.testES6TemplateLiteralsAreNotDirectives = function (test) {
+  var src = [
+    "function fn() {",
+    "`use strict`;",
+    "return \"\\077\";",
+    "}"
+  ];
+
+  TestRun(test)
+    .addError(2, "Expected an assignment or function call and instead saw an expression.")
```

Comment:
Another case we probably want to check is this:

```js
function fn() {
  `${"use strict"}`;
  return "something-strict-mode-only";
}
```

That will pass, but it's a nice test to have anyways